ethcore/src/executive.rs

```diff
@@ -410,7 +410,7 @@ impl<'a, B: 'a + StateBackend> Executive<'a, B> {
 			let default = [];
 			let data = if let Some(ref d) = params.data { d as &[u8] } else { &default as &[u8] };
 
-			let trace_info = tracer.prepare_trace_call(&params);
+			let trace_info = tracer.prepare_trace_call(&params, self.depth == 0);
```

Comment:
I'm not sure about the input. I think it's not cheaper than calling regular contract, is it? Why only trace input for regular contracts and not builtins then?

Reply:
I would keep the input then, since if we don't consider that an issue on a contract level we probably should also keep the same behaviour for builtins as well - the cost is payed up front by allocating the memory, so I think the amount of data that can be generated in traces is acceptable.